src/tsickle.ts

```diff
@@ -1691,7 +1702,12 @@ class ExternsWriter extends ClosureRewriter {
         jsdocTags.push({tagName: 'record'});
         jsdocTags.push({tagName: 'struct'});
       }
-      if (writeJsDoc) this.emit(jsdoc.toString(jsdocTags));
+      if (writeJsDoc) {
+        jsdocTags.push(SUPPRESS_DUPLICATE_TAG);
+        this.emit(jsdoc.toString(jsdocTags));
+      } else {
+        this.emit(jsdoc.toString([SUPPRESS_DUPLICATE_TAG]));
```

Comment:
This is weird, why are we emitting jsdoc if the flag is false?

I don't quite follow the logic above, but I think maybe we should drop the writeJSDoc flag and instead have the code above clear the jsdocTags array (?)

Reply:
Looked into it more, and when there is a constructor, we emit the jsdoc via `emitFunctionType` in this function. So we should actually add the suppress docs to that call. Changed the code to do so.